packages/connect/src/interceptor.spec.ts

```diff
@@ -78,38 +80,72 @@ const TestService = {
-    "inner sending request with headers: unary-request-header",
-    "inner response received with headers: unary-response-header",
+    "outer sending request with headers: connect-protocol-version, content-type, unary-request-header",
+    "inner sending request with headers: connect-protocol-version, content-type, unary-request-header",
+    "inner response received with headers: content-length, content-type, unary-response-header",
     "inner response done with trailers: unary-response-trailer",
-    "outer response received with headers: unary-response-header",
+    "outer response received with headers: content-length, content-type, unary-response-header",
     "outer response done with trailers: unary-response-trailer",
   ] as const;
+
   it("promise client", async () => {
     const log: string[] = [];
-    const transport = stubTransport({
-      interceptors: [
-        makeLoggingInterceptor("outer", log),
-        makeLoggingInterceptor("inner", log),
-      ],
-      unaryResponseHeader: new Headers({
-        "unary-response-header": "foo",
-      }),
-      unaryResponseTrailer: new Headers({
-        "unary-response-trailer": "foo",
-      }),
-    });
+    const input = { value: 9001 };
+    const output = new StringValue({ value: "output" });
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars -- necessary for the matcher's types
+    const fakeUnary = (_input: Int32Value, context: HandlerContext) => {
+      context.responseHeader.set("unary-response-header", "foo");
+      context.responseTrailer.set("unary-response-trailer", "foo");
+      return output;
+    };
+    const unary = jasmine.createSpy("unary", fakeUnary).and.callFake(fakeUnary);
+    const transport = createRouterTransport(
+      ({ service }) => {
+        service(TestService, { unary });
+      },
+      {
+        transport: {
+          interceptors: [
+            makeLoggingInterceptor("outer", log),
+            makeLoggingInterceptor("inner", log),
+          ],
+        },
+      }
+    );
 
-    await transport.unary(
+    const response = await transport.unary(
       TestService,
       TestService.methods.unary,
       undefined,
       undefined,
       {
         "unary-request-header": "foo",
       },
-      {}
+      input
     );
+    expect(unary).toHaveBeenCalledOnceWith(new Int32Value(input), {
+      method: TestService.methods.unary,
+      requestHeader: new Headers({
+        "content-type": "application/proto",
+        "connect-protocol-version": "1",
+        "unary-request-header": "foo"
+      }),
+      responseHeader: jasmine.any(Headers),
+      // Need help: cannot make sense of the jasmine output to make this work
+      // responseHeader: new Headers({
+      //   "content-type": "application/proto",
+      //   "unary-response-header": "foo"
+      // }),
+      responseTrailer: jasmine.any(Headers),
+      // Need help: cannot make sense of the jasmine output to make this work
```

Comment:
I added this in a separate commit so we can drop it if you don't have any ideas, but @timostamm I was not able to make sense of Jasmine's output here:

```
    Expected spy unary to have been called only once, and with given args:
      [ Int32Value({ value: 9001 }), Object({ method: Object({ name: 'Unary', I: Function, O: Function, kind: 0 }), requestHeader: [object Headers], responseHeader: <jasmine.any(Headers)>, responseTrailer: [object Headers], service: Object({ typeName: 'handwritten.TestService', methods: Object({ unary: Object({ name: 'Unary', I: Function, O: Function, kind: 0 }), serverStream: Object({ name: 'ServerStream', I: Function, O: Function, kind: 1 }) }) }) }) ]
    But the actual call was:
      [ Int32Value({ value: 9001 }), Object({ method: Object({ name: 'Unary', I: Function, O: Function, kind: 0 }), service: Object({ typeName: 'handwritten.TestService', methods: Object({ unary: Object({ name: 'Unary', I: Function, O: Function, kind: 0 }), serverStream: Object({ name: 'ServerStream', I: Function, O: Function, kind: 1 }) }) }), requestHeader: [object Headers], responseHeader: [object Headers], responseTrailer: [object Headers] }) ].
    Expected $[1].responseTrailer[Symbol(headers list)][Symbol(headers map sorted)] = [ [ 'unary-response-trailer', 'foo' ] ] to equal null.
```

any suggestions?

Reply:
It seems like it gives us pretty substantial value because it validates that the right headers were actually passed around, but if you aren't sure how to do it, I officially give up :upside_down_face:.  I tried pretty hard and it makes no sense to me why it fails.  Reverted in e438a394b41095362c64873b6dc4410ddeb84e60.

Thanks for taking a look.  It was worth a try to see how users out there in the wild might use this stuff.